ERROR: "__bad_udelay" [drivers/media/dvb/ttpci/budget.ko] undefined!
make[1]: *** [__modpost] Error 1

with CONFIG_DVB_BUDGET=m

diff --git a/drivers/media/dvb/ttpci/Kconfig b/drivers/media/dvb/ttpci/Kconfig
index ae88243..4b905b2 100644
--- a/drivers/media/dvb/ttpci/Kconfig
+++ b/drivers/media/dvb/ttpci/Kconfig
@@ -65,7 +65,7 @@ config DVB_AV7110_OSD
 
 config DVB_BUDGET_CORE
 	tristate "SAA7146 DVB cards (aka Budget, Nova-PCI)"
-	depends on DVB_CORE && PCI && I2C
+	depends on DVB_CORE && PCI && I2C  && (!ARM || BROKEN)
 	select VIDEO_SAA7146
 	select TTPCI_EEPROM
 	help
